common/archiver/s3store/historyArchiver_test.go

```diff
@@ -104,7 +104,7 @@ func (s *historyArchiverSuite) SetupTest() {
 	s.Assertions = require.New(s.T())
 	s.container = &archiver.HistoryBootstrapContainer{
 		Logger:        log.NewNoopLogger(),
-		MetricsClient: metrics.NewClient(scope, metrics.HistoryArchiverScope),
+		MetricsClient: metrics.NewClient(&metrics.ClientConfig{}, scope, metrics.HistoryArchiverScope),
```

Comment:
instead of passing in pointer to empty struct, we should make it optional and support passing nil and use default cfg if it is nil.

Reply:
Ideally we want to minimize amount of nil checks around the code especially if we go DI route. Also, this is a test code and we should not adjust behavior for test code explicitly unless it is unavoidable.